### PR TITLE
IBP-4474 no change password when user logged in for the first time

### PR DESCRIPTION
**Pull request by cuenyad**
Please, review.

### CODE REVIEW

File: src/main/java/org/ibp/api/domain/user/UserProfileDto.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
+package org.ibp.api.domain.user;
+
+import org.pojomatic.Pojomatic;
+import org.pojomatic.annotations.AutoProperty;
+
+@AutoProperty
+public class UserProfileDto {
```
Comment (nahuel-soldevilla) [inline src/main/java/org/ibp/api/domain/user/UserProfileDto.java:7]: Maybe `UserProfileUpdateRequestDTO` ?

File: src/main/java/org/ibp/api/java/impl/middleware/manager/UserValidator.java [hunk 2]
```diff
@@ -336,15 +336,52 @@ protected void validateUserRoles(final UserDetailDto user) {
...
+	}
+
+	public void validateUserProfileLogged(final BindingResult errors, final WorkbenchUser workbenchUser){
+		final WorkbenchUser loggedInUser = this.securityService.getCurrentlyLoggedInUser();
+		if (!workbenchUser.getUserid().equals(loggedInUser.getUserid())) {
+			errors.reject("user.perfile.update.invalid.user.logged", new Object[] {}, "");
+			throw new ApiRequestValidationException(errors.getAllErrors());
+
+		}
+	}
```
Comment (nahuel-soldevilla) [inline src/main/java/org/ibp/api/java/impl/middleware/manager/UserValidator.java:382]: I know you mentioned it but I forgot. What was the reason to reeive the username  in the request and then validate it? can we omit the whole thing? just use the getCurrentlyLoggedInUser when validating the password?

File: src/main/java/org/ibp/api/rest/user/profile/UserProfileResource.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	@Autowired
+	private UserService userService;
+
+	@Autowired
+	private org.generationcp.middleware.service.api.user.UserService userServiceMiddleware;
+
+	@Autowired
+	private AuthenticationManager authenticationManager;
+
+	@ApiOperation(value = "Update user profile", notes = "Update First name, Laste name and Email from user profile in this deployment instance of BMSAPI. ")
```
Comment (nahuel-soldevilla) [inline src/main/java/org/ibp/api/rest/user/profile/UserProfileResource.java:35]: typo _Laste_

File: src/main/resources/messages_en.properties [hunk 4]
```diff
@@ -423,6 +423,7 @@ users.can.not.be.edited.multiple.persons=User has duplicated person details. Con
 users.user.can.not.be.deactivated=Logged in user cannot be auto-deactivated
 users.update.superadmin.not.allowed=Updating user with superadmin role is not allowed
 
+user.perfile.update.invalid.user.logged=Update invalid, the user sent is not the same as the one logged in.
```
Comment (nahuel-soldevilla) [inline src/main/resources/messages_en.properties:426]: type _perfile_, but see if we can remove the whole username validation

File: src/main/java/org/ibp/api/rest/user/profile/UserProfileResource.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	@RequestMapping(value = "/users/profile", method = RequestMethod.PATCH)
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/rest/user/profile/UserProfileResource.java:36]: I will use my-profile instead

File: src/main/java/org/ibp/api/rest/user/profile/UserProfileResource.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+	@ResponseBody
+	public ResponseEntity<Integer> updateProfile(
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/rest/user/profile/UserProfileResource.java:38]: why do we need to return integer?

File: src/main/java/org/ibp/api/domain/user/UserProfileDto.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+	private String userName;
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/domain/user/UserProfileDto.java:9]: I would not send the userName, we are changing the profile for the logged in user, so we can get the user from the session, get the username from there and validate the password

File: src/main/java/org/ibp/api/java/impl/middleware/manager/UserValidator.java (hunk 2 above)
Comment (clarysabel) [reply src/main/java/org/ibp/api/java/impl/middleware/manager/UserValidator.java:382]: Agree, I left a similar comment in the Dto, so we dont need to send the username

**Pull request merged**